usl_models/usl_models/flood_ml/prediction_dataset.py

```diff
@@ -0,0 +1,195 @@
+"""tf.data.Datasets for training FloodML model on CityCAT data."""
+
+import logging
+import random
+from typing import Iterator, Optional, Tuple
+import urllib.parse
+
+from google.cloud import firestore  # type:ignore[attr-defined]
+from google.cloud import storage  # type:ignore[attr-defined]
+import numpy
+from numpy.typing import NDArray
+import tensorflow as tf
+
+from usl_models.flood_ml import constants
+from usl_models.flood_ml import metastore
+from usl_models.flood_ml import model
+
+
+def load_prediction_dataset(
+    study_area: str,
+    city_cat_config: str, 
+    batch_size: int = 4,
+    n_flood_maps: int = constants.N_FLOOD_MAPS,
+    m_rainfall: int = constants.M_RAINFALL,
+    max_chunks: Optional[int] = None,
+    firestore_client: Optional[firestore.Client] = None,
+    storage_client: Optional[storage.Client] = None,
+) -> tf.data.Dataset:
+    """Creates a prediction dataset which generates chunks for flood model prediction.
+
+    This dataset produces the input for `model.call_n`.
+    For training with teacher-forcing, `load_dataset_windowed` should be used instead.
+    The examples are generated from multiple simulations.
+    The dataset iteratively yields examples read from Google Cloud Storage to avoid
+    pulling all examples into memory at once.
+
+    Args:
+      
+      batch_size: Size of batches yielded by the dataset. Approximate memory
+                  usage is 10GB * batch_size during training.
+      n_flood_maps: The number of flood maps in each example.
+      m_rainfall: The width of the temporal rainfall tensor.
+      max_chunks: The maximum number of examples to yield from the dataset.
+                  If `None` (default) yields all examples from the simulations.
+      firestore_client: The client to use when interacting with Firestore.
+      storage_client: The client to use when interacting with Cloud Storage.
+    """
+    firestore_client = firestore_client or firestore.Client()
+    storage_client = storage_client or storage.Client()
+
+    def generator():
+        """Generator for producing full inputs from study area"""
+        for model_input, metadata in _iter_model_inputs_for_prediction(
+        firestore_client,
```

Comment:
These should be indented. Running [black](https://github.com/psf/black) should take care of this for you.

Reply:
I have black running but maybe i need to run  `Format` specifically but good to know.

Reply:
I figured out the issue with black. I switched VM to use tf2.14 and that didn't install black in VSCode's remote SSH session. I was wondering why I was not harasses by it ;)